ui/src/components/SettingsDialog.tsx

```diff
@@ -0,0 +1,104 @@
+import { useDismissNavigate } from '@/logic/routing';
+import { SettingsState, useCalm, useSettingsState } from '@/state/settings';
+import { isTalk } from '@/logic/utils';
+import Dialog from './Dialog';
+import Setting from './Setting';
+
+async function toggle(property: keyof SettingsState['calmEngine']) {
+  const selProp = (s: SettingsState) => s.calmEngine[property];
+  const state = useSettingsState.getState();
+  const curr = selProp(state);
+  await state.putEntry('calmEngine', property, !curr);
```

Comment:
I think putEntry uses poke under the hood, if we switched it to `trackedPoke` progress would be a little more accurate.